ad/views.py

```diff
@@ -29,6 +30,27 @@ class ModelMappingMixin:
         'class': ClassAdForm,
     }
 
+    def get_model(self):
+        adtype = self.kwargs['adtype']
+        model = self.model_mapping.get(adtype)
+        if not model:
+            raise Http404(f"No model found for ad type: {adtype}")
+        return model
+
+    def get_form_class(self):
+        adtype = self.kwargs['adtype']
+        return self.form_mapping.get(adtype)
+
+    def get_queryset(self):
+        model = self.get_model()
+        return model.objects.all()
+    
+    def get_object(self, queryset=None):
+        obj = super().get_object(queryset)
+        if obj.owner != self.request.user:
+            raise Http404("You do not have permission to edit this ad.")
+        return obj
```

Comment:
This will not be needed if you modify get_queryset() method to filter current user ads only. It will automatically raise 404 error if object is not found in queryset.